src/main/kotlin/com/hypto/iam/server/Application.kt

```diff
@@ -89,8 +89,14 @@ import org.koin.ktor.plugin.Koin
 import org.koin.logger.SLF4JLogger
 
 private const val REQUEST_ID_HEADER = "X-Request-Id"
-private val shutdownHook = EngineShutdownHook(1.seconds, 1.minutes, 5.minutes)
+
+private val preWait = 1.seconds
+private val gracePeriod = 30.seconds
+private val timeout = 45.seconds
+private val shutdownHook = EngineShutdownHook(preWait, gracePeriod, timeout)
+
```

Comment:
nit: If the point of this change is readability, in kotlin, this can be done as
```
private val shutdownHook = EngineShutdownHook(preWait = 1.seconds, gracePeriod = 30.seconds, timeout = 45.seconds)
```

Reply:
Makes sense. It was readability, yes. I was also wondering if people could just check the function definition and find out the parameters. But yeah, named parameters helps with making less mistakes, given all are of same type of data so it's just a matter of order of the data and people might change order